### PR TITLE
Create role feature is incorporated into the B2B sample application

### PR DESCRIPTION
**Pull request by Achintha444**
## Purpose
Developed the create role feature in the B2B sample application. UI of this feature is given below.

Screenshot 2022-11-15 at 14 35 27
Screenshot 2022-11-15 at 14 35 37
Screenshot 2022-11-15 at 14 35 44

### CODE REVIEW

File: b2b-sample/guardio-insurance-sample-application/components/sections/sections/settingsSection/roleManagementSection/otherComponents/createRoleComponent/createRoleComponentInner/permission.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+                            <div className="buttons">
+                                <FormSuite.Group>
+                                    <ButtonToolbar>
+                                        <Button
+                                            className={ styles.addUserButton }
+                                            size="lg"
+                                            appearance="ghost"
+                                            type="submit"
+                                            onClick={ onPrevious }>
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/components/sections/sections/settingsSection/roleManagementSection/otherComponents/createRoleComponent/createRoleComponentInner/permission.js:82]:
```suggestion
                                            onClick={ onPrevious }
                                        >
```

Check the other places as well.

File: b2b-sample/guardio-insurance-sample-application/components/sections/sections/settingsSection/roleManagementSection/roleManagementSectionComponent.js [hunk 2]
```diff
@@ -52,27 +55,30 @@ export default function RoleManagementSectionComponent(prop) {
...
+                session={ session } />
             {
                 rolesList
                     ? <RolesList session={ session } rolesList={ rolesList } />
                     : (<EmptySettings
                         bodyString="There are no roles created for the organization."
                         buttonString="Create role"
                         icon={ <PeoplesIcon style={ { opacity: .2 } } width="150px" height="150px" /> }
-                        onAddButtonClick={ ()=>{} }
+                        onAddButtonClick={ () => { } }
```
Comment (thivi) [inline b2b-sample/guardio-insurance-sample-application/components/sections/sections/settingsSection/roleManagementSection/roleManagementSectionComponent.js:81]: Why is this an empty function?

**Review by thivi — DISMISSED** [review]

File: b2b-sample/guardio-insurance-sample-application/components/sections/sections/settingsSection/roleManagementSection/roleManagementSectionComponent.js (hunk 2 above)
Comment (Achintha444) [reply b2b-sample/guardio-insurance-sample-application/components/sections/sections/settingsSection/roleManagementSection/roleManagementSectionComponent.js]: Fixed

**Review by thivi — APPROVED** [review]

File: b2b-sample/guardio-insurance-sample-application/components/sections/sections/settingsSection/roleManagementSection/otherComponents/createRoleComponent/createRoleComponent.js [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+     * 
+     * @param nextStep - next step
+     */
+    const onChange = (nextStep) => {
+        setStep(nextStep < 0 ? 0 : nextStep > 3 ? 3 : nextStep);
+    };
+    const onPrevious = () => onChange(step - 1);
+    const onNext = () => {
+        if (step == 2) {
+
```
Comment (lashinijay) [inline b2b-sample/guardio-insurance-sample-application/components/sections/sections/settingsSection/roleManagementSection/otherComponents/createRoleComponent/createRoleComponent.js:52]: make it consistent through out the code to leave a new line after the if condition or not.

File: b2b-sample/guardio-insurance-sample-application/components/sections/sections/settingsSection/roleManagementSection/otherComponents/createRoleComponent/createRoleComponent.js [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+    const onCreateRoleModalClose = () => {
+        setOpenCreateRoleModal(false);
+        setStep(0);
+        setDisplayName("");
+        setPermissions([]);
+        setUsers([]);
+    };
+
+    const craeteRoleItemDetailsComponent = (currentStep) => {
+
```
Comment (lashinijay) [inline b2b-sample/guardio-insurance-sample-application/components/sections/sections/settingsSection/roleManagementSection/otherComponents/createRoleComponent/createRoleComponent.js:89]: make this also consistent. leave a line after the declaration.

File: b2b-sample/guardio-insurance-sample-application/pages/api/settings/role/createRole.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+ * under the License.
+ */
+
+import { getSentDataRequestOptions } from "../../../../util/util/apiUtil/getSentDataRequestOptions";
+import { dataNotRecievedError, notPostError } from "../../../../util/util/apiUtil/localResErrors";
+import { RequestMethod } from "../../../../util/util/apiUtil/requestMethod";
+import { getOrgUrl } from "../../../../util/util/orgUtil/orgUtil";
+
+/**
+ * backend API call to list all roles.
```
Comment (lashinijay) [inline b2b-sample/guardio-insurance-sample-application/pages/api/settings/role/createRole.js:25]: Let's start the comments with upper case letter and ends with a full-stop. Check other places too.

Comment (Achintha444) [reply b2b-sample/guardio-insurance-sample-application/pages/api/settings/role/createRole.js:25]: Fixed with https://github.com/wso2/samples-is/pull/416/commits/8b059f39358c5652ea99e473f4cf34e23ad95e0b

File: b2b-sample/guardio-insurance-sample-application/components/sections/sections/settingsSection/roleManagementSection/otherComponents/createRoleComponent/createRoleComponent.js (hunk 3 above)
Comment (Achintha444) [reply b2b-sample/guardio-insurance-sample-application/components/sections/sections/settingsSection/roleManagementSection/otherComponents/createRoleComponent/createRoleComponent.js:52]: Fixed with https://github.com/wso2/samples-is/pull/416/commits/e535216c72dd635ee40f158f18eab8bdfffa3585

File: b2b-sample/guardio-insurance-sample-application/components/sections/sections/settingsSection/roleManagementSection/otherComponents/createRoleComponent/createRoleComponent.js (hunk 4 above)
Comment (Achintha444) [reply b2b-sample/guardio-insurance-sample-application/components/sections/sections/settingsSection/roleManagementSection/otherComponents/createRoleComponent/createRoleComponent.js:89]: Fixed with https://github.com/wso2/samples-is/pull/416/commits/e535216c72dd635ee40f158f18eab8bdfffa3585

**Pull request merged**